lib/EnzymeCore/src/EnzymeCore.jl

```diff
@@ -421,6 +421,8 @@ Return a new instance of [`ReverseModeSplit`](@ref) mode where `Width` is set to
 @inline set_runtime_activity(::ReverseModeSplit{ReturnPrimal,ReturnShadow,RuntimeActivity,Width,ModifiedBetween,ABI, Holomorphic, ErrIfFuncWritten, ShadowInit}, rt::Bool) where {ReturnPrimal,ReturnShadow,RuntimeActivity,Width,ModifiedBetween,ABI, Holomorphic, ErrIfFuncWritten, ShadowInit} = ReverseModeSplit{ReturnPrimal,ReturnShadow,rt,Width,ModifiedBetween,ABI, Holomorphic, ErrIfFuncWritten, ShadowInit}()
 @inline clear_runtime_activity(::ReverseModeSplit{ReturnPrimal,ReturnShadow,RuntimeActivity,Width,ModifiedBetween,ABI, Holomorphic, ErrIfFuncWritten, ShadowInit}) where {ReturnPrimal,ReturnShadow,RuntimeActivity,Width,ModifiedBetween,ABI, Holomorphic, ErrIfFuncWritten, ShadowInit} = ReverseModeSplit{ReturnPrimal,ReturnShadow,false,Width,ModifiedBetween,ABI, Holomorphic, ErrIfFuncWritten, ShadowInit}()
 
+@inline set_abi(::ReverseModeSplit{ReturnPrimal,ReturnShadow,RuntimeActivity,Width,ModifiedBetween,OldABI,Holomorphic,ErrIfFuncWritten,ShadowInit}, ::Type{NewABI}) where {ReturnPrimal,ReturnShadow,RuntimeActivity,Width,ModifiedBetween,OldABI,Holomorphic,ErrIfFuncWritten,ShadowInit,NewABI<:ABI} = ReverseModeSplit{ReturnPrimal,ReturnShadow,RuntimeActivity,Width,ModifiedBetween,NewABI,Holomorphic,ErrIfFuncWritten,ShadowInit}()
```

Comment:
just noticed this is also missing a version for set_abi on the type [not just value], otherwise lgtm

Reply:
I'm not sure what you mean, the signature is
```
set_abi(::Mode, ::Type{A}) where {A<:ABI}
```
and it is implemented consistently for all modes. Do you mean you also want
```
set_abi(::Type{M}, ::Type{A}) where {M<:Mode,A<:ABI}
```
everywhere?

Reply:
like this https://github.com/EnzymeAD/Enzyme.jl/commit/03d70353863e245ceae1dc04a201575073509be2

Reply:
you add set_abi methods for split, and it should come with both type and mode

Reply:
done